mkosi/log.py

```diff
@@ -6,8 +6,10 @@
 ARG_DEBUG: set[str] = set()
 
 
-def die(message: str, exception: Optional[Exception] = None) -> NoReturn:
+def die(message: str, exception: Optional[Exception] = None, hint: Optional[str] = None) -> NoReturn:
```

Comment:
I made the second change, but not the first. I think using `""` is an implicit false value is smelly. We need to treat it in a special way anyway, and I think using None for that is clearer.

Reply:
I can live with that, but I do think that using the empty string to encode this is actually cleaner, since it's already an "illegal" value or rather the empty message is the trivial null message. In the end, it type checks, nobody sees the difference from the outside, so it's fine. :)